crates/objc2-encode/src/encoding.rs

```diff
@@ -248,6 +251,39 @@ impl Encoding {
     pub fn equivalent_to_box(&self, other: &EncodingBox) -> bool {
         compare_encodings(self, other, NestingLevel::new(), false)
     }
+
+    /// Computes the theoretical size in bytes of the represented value type,
+    /// considering everything is packed, i.e. no alignment is computed.
+    pub fn size(&self) -> usize {
+        Helper::new(self).size(NestingLevel::new())
+    }
+
+    /// Computes the raw signature string of the object corresponding to the
+    /// block taking `args` as inputs and returning `ret`.
+    ///
+    /// See [`objc2::encode::block_signature_string`] for more details. Prefer
+    /// using it whenever possible.
+    ///
+    /// [`objc2::encode::block_signature_string`]: https://docs.rs/objc2/latest/objc2/encode/fn.block_signature_string.html
+    pub fn block_signature_string(args: &[Self], ret: &Self) -> String {
+        // TODO: alignment?
```

Comment:
Raising this as a comment on the PR for visibility, I'm not sure what to do about alignment myself.